lib/BlockedBy.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const NodeResque = require('node-resque');
+const { queuePrefix } = require('../config/redis');
+
+class BlockedBy extends NodeResque.Plugin {
+    async beforePerform() {
+        const { jobId, blockedBy } = await this.jobInfo();
+        const blockingJobKeys = await this.queueObject.connection.redis.mget(blockedBy);
+        const blockingJobsRunning = blockingJobKeys.some(j => j !== null);
+
+        // If any blocking job is running, then re-enqueue
+        if (blockingJobsRunning) {
+            await this.reEnqueue();
+
+            return false;
+        }
+
+        // Register the curent job as running by setting key
+        // Set expire time to take care of the case where
+        // afterPerform failed to call and blocked jobs will be stuck forever
+        await this.queueObject.connection.redis.expire(jobId, this.lockTimeout());
```

Comment:
Curious, does `expire` do a set and expire or just expire on an existing key?

Reply:
It will only expire. I forgot to set :p